internal/repository/employee/employee_mysql.go

```diff
@@ -0,0 +1,141 @@
+package employee
+
+import (
+	"ProyectoFinal/pkg/models"
+	"database/sql"
+	"fmt"
+)
+
+type mysqlRepository struct {
+	db *sql.DB
+}
+
+func NewMySQLRepository(db *sql.DB) Repository {
+	return &mysqlRepository{
+		db: db,
+	}
+}
+
+func (r *mysqlRepository) GetAll() ([]models.Employee, error) {
+	rows, err := r.db.Query(QueryGetAllEmployees)
+	if err != nil {
+		return nil, fmt.Errorf("error querying employees: %w", err)
+	}
+	defer rows.Close()
+
+	var employees []models.Employee
+	for rows.Next() {
+		var employee models.Employee
+		var warehouseID sql.NullInt64
+
+		err := rows.Scan(&employee.ID, &employee.CardNumberID, &employee.FirstName, &employee.LastName, &warehouseID)
+		if err != nil {
+			return nil, fmt.Errorf("error scanning employee row: %w", err)
+		}
+
+		if warehouseID.Valid {
```

Comment:
Si en la estructura de Employee pones el campo warehouseID como opcional (con puntero), no es necesario hacer la validación del .Valid, ni tener la variable auxiliar warehouseID

Creo que seria en las siguientes 3 estructuras :

```suggestion
		type Employee struct {
	ID           int
	CardNumberID string
	FirstName    string
	LastName     string
	WarehouseID * int
}
type EmployeeDoc struct {
	ID           int    `json:"id"`
	CardNumberID string `json:"card_number_id"`
	FirstName    string `json:"first_name"`
	LastName     string `json:"last_name"`
	WarehouseID  *int    `json:"warehouse_id"`
}
type EmployeeRequest struct {
	CardNumberID string `json:"card_number_id" validate:"required"`
	FirstName    string `json:"first_name" validate:"required"`
	LastName     string `json:"last_name" validate:"required"`
	WarehouseID * int    `json:"warehouse_id" validate:"omitempty,gt=0"`
}
```